ofctrl/ofSwitch.go

```diff
@@ -342,17 +342,20 @@ func (self *OFSwitch) handleMessages(dpid net.HardwareAddr, msg util.Message) {
 
 	case *openflow15.MultipartReply:
 		log.Debugf("Received MultipartReply")
-		rep := (*openflow15.MultipartReply)(t)
-		if self.monitorEnabled {
-			key := fmt.Sprintf("%d", rep.Xid)
-			ch, found := monitoredFlows.Get(key)
-			if found {
-				replyChan := ch.(chan *openflow15.MultipartReply)
-				replyChan <- rep
+		switch t.Type {
+		case openflow15.MultipartType_FlowDesc:
+			if self.monitorEnabled {
```

Comment:
In my opinion, the following code would be a bit safer:
```go
key := fmt.Sprintf("%d", t.Xid)
ch, found := monitoredFlows.Get(key)
if found {
        if self.monitorEnabled {
                replyChan := ch.(chan *openflow15.MultipartReply)
                replyChan <- rep
        }
        monitoredFlows.Remove(key)
}
```

this way we make sure that the key is removed from the map if present, regardless of the value of `self.monitorEnabled`. I prefer it because the rest of the code doesn't really offer strong guarantees, such as `DumpFlowStats` only being called when `self.monitorEnabled` is true.

Reply:
OK, updated.

Reply:
the latest version is not what I suggested
I was suggesting checking for existence of the key even if `self.monitorEnabled` is false, to ensure proper cleanup (also, ideally, entries in the map that are not removed should be expired after some time, in case no reply is ever received?).